docs/reference/builder.md

```diff
@@ -953,6 +953,41 @@ For example:
 The output of the final `pwd` command in this `Dockerfile` would be
 `/path/$DIRNAME`
 
+## ARG
+    ARG [--default=<value>] [--description=<value>] <arg name>
+
+The `ARG` instruction defines a variable that is allowed to be passed as
+build-time variable to the builder. It is used to define a single variable at a time.
+Multiple variables may be defined by specifying `ARG` more than once. The `ARG`
+instruction can optionally take default value and description of the variable
+being defined as well. These are persisted in the image history so they can be
```

Comment:
s/history/metadata/ since I think its better to use "docker inspect" instead of "docker history" for this info.